.ci/targets/ios_platform_tests.yaml

```diff
@@ -22,3 +22,5 @@ tasks:
     # So we run `drive-examples` after `native-test`; changing the order will result ci failure.
     script: script/tool_runner.sh
     args: ["drive-examples", "--ios", "--exclude=script/configs/exclude_integration_ios.yaml"]
+  - name: remove simulator
+    script: .ci/scripts/remove_simulator.sh
```

Comment:
Should this be run in an `always:` block? https://cirrus-ci.org/guide/writing-tasks/

Reply:
Ha never mind this is luci, is there a cleanup mechanism in this fancy yaml system that moved everything out of the recipe?

Reply:
> is there a cleanup mechanism in this fancy yaml system that moved everything out of the recipe?

You may be overestimating the level of fanciness here :)

Now that you mention it, I don't actually know if currently it'll stop on the first failing step, or run everything. I should upload a build-breaking PR and see what runs.

Reply:
Answer: no, there's no cleanup mechanism 😬 

I filed https://github.com/flutter/flutter/issues/122629. Should we land this as-is to at least slow the bleeding of simulators, and then adjust it later once we have a cleanup stage?